fairseq/logging/progress_bar.py

```diff
@@ -383,6 +383,75 @@ def _log_to_tensorboard(self, stats, tag=None, step=None):
         writer.flush()
 
 
+try:
+    import aim
+except ImportError:
+    aim = None
+
+
+class AimProgressBarWrapper(BaseProgressBar):
+    """Log to Aim"""
+
+    def __init__(self, wrapped_bar, aim_repo_path, aim_experiment_name,
+                 system_tracking_interval=None):
+        self.wrapped_bar = wrapped_bar
+        if aim is None:
+            logger.warning("aim not found, pip install aim")
+            return
+
+        from aim.sdk.session.session import Session
+        from aim.engine.configs import DEFAULT_SYSTEM_TRACKING_INT
+        from aim.engine.utils import convert_to_py_number
+        self._convert_to_py_number = convert_to_py_number
+
+        self._system_tracking_interval = system_tracking_interval
+        if system_tracking_interval is None:
+            self._system_tracking_interval = DEFAULT_SYSTEM_TRACKING_INT
+
+        self._experiment_name = aim_experiment_name
+        self._aim_session = Session(
+            repo=aim_repo_path,
+            experiment=self._experiment_name,
+            system_tracking_interval=self._system_tracking_interval,
+        )
+
+    def __iter__(self):
+        return iter(self.wrapped_bar)
+
+    def log(self, stats, tag=None, step=None):
+        """Log intermediate stats to Aim."""
+        self._log_to_aim(stats, tag, step)
+        self.wrapped_bar.log(stats, tag=tag, step=step)
+
+    def print(self, stats, tag=None, step=None):
+        """Print end-of-epoch stats."""
+        self._log_to_aim(stats, tag, step)
+        self.wrapped_bar.print(stats, tag=tag, step=step)
+
+    def update_config(self, config):
+        """Log latest hparams."""
+        self._aim_session.set_params(config, name="hparams")
+        self.wrapped_bar.update_config(config)
+
+    def _log_to_aim(self, stats, tag=None, step=None):
+        if aim is None:
+            return
+        if step is None:
+            step = stats["num_updates"]
+
+        prefix = "" if tag is None else tag
+
+        for key in stats.keys() - {"num_updates"}:
+            value = stats[key]
+            if isinstance(stats[key], AverageMeter):
+                value = stats[key].val
+            value = self._convert_to_py_number(value)
```

Comment:
No need to convert value here, because it will be converted inside the `track` function.